### PR TITLE
database: Clean up function definitions

### PR DESCRIPTION
**Pull request by RobLoach**
Addresses the comments made in https://github.com/libretro/RetroArch/pull/11719#pullrequestreview-824570173 .

### CODE REVIEW

File: libretro-common/string/stdstring.c [hunk 1]
```diff
@@ -631,87 +630,3 @@ int find_index_substring_string(const char* str1, const char* str2)
 
    return -1;
 }
-
```
Comment (pkos) [inline libretro-common/string/stdstring.c:634]: Why is this removed?  Its needed for multi disc support.

Comment (pkos) [reply libretro-common/string/stdstring.c:634]: Oh, these were intentionally broken out from the string functions and placed back in the tasks.  Roger.

**Review by pkos — APPROVED** [review]
Nicely done cleaning up.

File: libretro-common/string/stdstring.c (hunk 1 above)
Comment (RobLoach) [reply libretro-common/string/stdstring.c:634]: Moved the 2 cue-related functions to the cue file, since it doesn't really fit in with the rest of the libretro-,common string functions. They're more serial scanning and manipulation.
